modules/terraform/format.go

```diff
@@ -50,11 +50,26 @@ func FormatArgs(options *Options, args ...string) []string {
 	lockSupported := collections.ListContains(TerraformCommandsWithLockSupport, commandType)
 	planFileSupported := collections.ListContains(TerraformCommandsWithPlanFileSupport, commandType)
 
+	includeVars := true
+	// we don't want to include any -var flags if the command is apply
+	// and plan file is provided
+	if options.PlanFilePath != "" && commandType == "apply" {
+		includeVars = false
+	}
```

Comment:
NIT: slightly fewer lights:

```suggestion
	// Include -var and -var-file flags unless we're running 'apply' with a plan file
	includeVars := !(commandType == "apply" && len(options.PlanFilePath) > 0)
```